complex/ft_msg.c

```diff
@@ -290,6 +309,48 @@ int ft_recv_dgram(void)
 	return -FI_ETIMEDOUT;
 }
 
+int ft_recv_dgram_flood(void)
+{
+	struct timespec s, e;
+	int credits, ret;
+	int64_t poll_time = 0;
+	int expected = ft.xfer_iter;
+	int got	     = 0;
+	do {
+		if (ft_rx.credits > (ft_rx.max_credits >> 1)) {
+			ret = ft_post_recv_bufs();
+			if (ret)
+				return ret;
+		}
+
+		credits = ft_rx.credits;
+
+		ret = ft_comp_rx();
+		if (credits != ft_rx.credits) {
+			int num_creds = ft_rx.credits - credits;
+			poll_time=0;
+			got+=num_creds;
+		}
+		if (ret)
+			return ret;
+
+		if(got >= expected) {
+			return 0;
```

Comment:
The receive side does not exit until it receives the exact number of sends requested for the test.  But the send side only sends that many.  If any messages are lost, the result of the test is that the send side will send its last message repeatedly, on a timeout, until the receiver has received the correct number.

This will result in lower performance than is expected.  Also, the sender only retries the last send 1000 times.  Some of the tests will send millions of messages.  If more than 1000 packets are dropped, the test will fail.

It would be better for the receiver to simple receive messages until it gets the 'done' message.  The performance data would then be based on the number of packets actually received.